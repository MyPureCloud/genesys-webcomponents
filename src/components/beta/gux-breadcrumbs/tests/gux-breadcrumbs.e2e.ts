import { newE2EPage } from '@stencil/core/testing';

describe('gux-breadcrumbs-beta', () => {
  [
    `<gux-breadcrumbs-beta lang="en"></gux-breadcrumbs-beta>`,
    `
      <gux-breadcrumbs-beta lang="en">
        <gux-breadcrumb-item>Category</gux-breadcrumb-item>
        <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
        <gux-breadcrumb-item>Current</gux-breadcrumb-item>
      </gux-breadcrumbs-beta>
    `,
    `
      <gux-breadcrumbs-beta lang="en">
        <gux-breadcrumb-item>Category</gux-breadcrumb-item>
        <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
        <gux-breadcrumb-item href="#">Current</gux-breadcrumb-item>
      </gux-breadcrumbs-beta>
    `,
    `
      <gux-breadcrumbs-beta lang="en" accent="primary">
        <gux-breadcrumb-item>Category</gux-breadcrumb-item>
        <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
        <gux-breadcrumb-item>Current</gux-breadcrumb-item>
      </gux-breadcrumbs-beta>
    `,
    `
      <gux-breadcrumbs-beta lang="en" accent="secondary">
        <gux-breadcrumb-item>Category</gux-breadcrumb-item>
        <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
        <gux-breadcrumb-item>Current</gux-breadcrumb-item>
      </gux-breadcrumbs-beta>
    `,
    `
      <gux-breadcrumbs-beta lang="en" accent="invalid">
        <gux-breadcrumb-item>Category</gux-breadcrumb-item>
        <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
        <gux-breadcrumb-item>Current</gux-breadcrumb-item>
      </gux-breadcrumbs-beta>
    `
  ].forEach((content, index) => {
    it(`should display component as expected (${index + 1})`, async () => {
      const page = await newE2EPage();

      await page.setContent(content);

      const element = await page.find('gux-breadcrumbs-beta');

      expect(element.outerHTML).toMatchSnapshot();
    });
  });
});
