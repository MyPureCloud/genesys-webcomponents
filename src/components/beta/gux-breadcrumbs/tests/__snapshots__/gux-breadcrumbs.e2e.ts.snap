// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-breadcrumbs-beta should display component as expected (1) 1`] = `"<gux-breadcrumbs-beta lang=\\"en\\" class=\\"hydrated\\"><!----><nav aria-label=\\"Breadcrumbs\\" class=\\"gux-breadcrumbs-container gux-large-text\\"></nav></gux-breadcrumbs-beta>"`;

exports[`gux-breadcrumbs-beta should display component as expected (2) 1`] = `"<gux-breadcrumbs-beta lang=\\"en\\" class=\\"hydrated\\"><!----><nav aria-label=\\"Breadcrumbs\\" class=\\"gux-breadcrumbs-container gux-large-text\\"> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><span class=\\"gux-breadcrumb-content\\">Category</span><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item href=\\"#\\" class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><span class=\\"gux-breadcrumb-content\\">Parent</span><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><span class=\\"gux-breadcrumb-content\\">Current</span></span></gux-breadcrumb-item> </nav></gux-breadcrumbs-beta>"`;

exports[`gux-breadcrumbs-beta should display component as expected (3) 1`] = `"<gux-breadcrumbs-beta lang=\\"en\\" class=\\"hydrated\\"><!----><nav aria-label=\\"Breadcrumbs\\" class=\\"gux-breadcrumbs-container gux-large-text\\"> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><span class=\\"gux-breadcrumb-content\\">Category</span><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item href=\\"#\\" class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><span class=\\"gux-breadcrumb-content\\">Parent</span><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item href=\\"#\\" class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><span class=\\"gux-breadcrumb-content\\">Current</span></span></gux-breadcrumb-item> </nav></gux-breadcrumbs-beta>"`;

exports[`gux-breadcrumbs-beta should display component as expected (4) 1`] = `"<gux-breadcrumbs-beta lang=\\"en\\" accent=\\"primary\\" class=\\"hydrated\\"><!----><nav aria-label=\\"Breadcrumbs\\" class=\\"gux-breadcrumbs-container gux-large-text\\"> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><span class=\\"gux-breadcrumb-content\\">Category</span><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item href=\\"#\\" class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><span class=\\"gux-breadcrumb-content\\">Parent</span><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><span class=\\"gux-breadcrumb-content\\">Current</span></span></gux-breadcrumb-item> </nav></gux-breadcrumbs-beta>"`;

exports[`gux-breadcrumbs-beta should display component as expected (5) 1`] = `"<gux-breadcrumbs-beta lang=\\"en\\" accent=\\"secondary\\" class=\\"hydrated\\"><!----><nav aria-label=\\"Breadcrumbs\\" class=\\"gux-breadcrumbs-container\\"> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><span class=\\"gux-breadcrumb-content\\">Category</span><gux-icon class=\\"gux-breadcrumb-separator hydrated\\" icon-name=\\"ic-chevron-small-right\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"> <path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M11 8L7 4 6 5l3 3-3 3 1 1z\\"></path> </svg></div></gux-icon></span></gux-breadcrumb-item> <gux-breadcrumb-item href=\\"#\\" class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><a class=\\"gux-breadcrumb-content\\" href=\\"#\\">Parent</a><gux-icon class=\\"gux-breadcrumb-separator hydrated\\" icon-name=\\"ic-chevron-small-right\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"> <path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M11 8L7 4 6 5l3 3-3 3 1 1z\\"></path> </svg></div></gux-icon></span></gux-breadcrumb-item> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><span class=\\"gux-breadcrumb-content\\">Current</span></span></gux-breadcrumb-item> </nav></gux-breadcrumbs-beta>"`;

exports[`gux-breadcrumbs-beta should display component as expected (6) 1`] = `"<gux-breadcrumbs-beta lang=\\"en\\" accent=\\"invalid\\" class=\\"hydrated\\"><!----><nav aria-label=\\"Breadcrumbs\\" class=\\"gux-breadcrumbs-container\\"> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><span class=\\"gux-breadcrumb-content\\">Category</span><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item href=\\"#\\" class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><a class=\\"gux-breadcrumb-content\\" href=\\"#\\">Parent</a><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation\\"><span class=\\"gux-breadcrumb-content\\">Current</span></span></gux-breadcrumb-item> </nav></gux-breadcrumbs-beta>"`;
